### PR TITLE
Signer and IDTokenCredentials implementation using default service account on GCE

### PR DESCRIPTION
**Pull request by kryzthov**
This branch includes:
 - a Signer implementation that relies on the IAM API ([signBlob](https://cloud.google.com/iam/reference/rest/v1/projects.serviceAccounts/signBlob)) using the default Credentials
 - a Credentials implementation that provides ID tokens, using the Signer above.

### CODE REVIEW

**Comment by theacodes** [comment]
Hi @kryzthov - just a heads up that I'm currently OOO until next week but I'll take a quick lookl

**Comment by kryzthov** [comment]
All good! No rush, it's been there for a while, and it's not blocking! Enjoy your time off!

File: google/auth/compute_engine/credentials.py [hunk 1]
```diff
@@ -108,3 +117,163 @@ def service_account_email(self):
     def requires_scopes(self):
         """False: Compute Engine credentials can not be scoped."""
         return False
+
+
+class Signer(crypt.Signer):
```
Comment (theacodes) [inline google/auth/compute_engine/credentials.py]: We already have exactly this class here: https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/master/google/auth/iam.py#L36

File: google/auth/compute_engine/credentials.py [hunk 2]
```diff
@@ -108,3 +117,163 @@ def service_account_email(self):
...
+
+
+_DEFAULT_TOKEN_LIFETIME_SECS = 3600  # 1 hour in seconds
+_DEFAULT_TOKEN_URI = 'https://www.googleapis.com/oauth2/v4/token'
+
+
+class IDTokenCredentials(credentials.Credentials, credentials.Signing):
+    """Open ID Connect ID Token-based service account credentials.
+
+    These credentials relies on the default service account of a GCE instance.
```
Comment (theacodes) [inline google/auth/compute_engine/credentials.py]: Could you get it to work with any of the service accounts on the GCE instance?

Comment (kryzthov) [reply google/auth/compute_engine/credentials.py]: Done

**Comment by kryzthov** [comment]
Anymore feedback on this?

File: google/auth/compute_engine/credentials.py [hunk 3]
```diff
@@ -108,3 +114,126 @@ def service_account_email(self):
...
+        super(IDTokenCredentials, self).__init__()
+
+        if service_account_email is None:
+            sa_info = _metadata.get_service_account_info(request)
+            service_account_email = sa_info['email']
+        self._service_account_email = service_account_email
+
+        self._signer = iam.Signer(
+            request=request,
+            credentials=Credentials(),
```
Comment (theacodes) [inline google/auth/compute_engine/credentials.py]: don't you need to pass the service account email in here as well?

**Pull request merged**